spring-cloud-gcp-autoconfigure/pom.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>spring-cloud-gcp</artifactId>
+        <groupId>org.springframework.cloud</groupId>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>spring-cloud-gcp-autoconfigure</artifactId>
+    <name>Spring Cloud GCP Autoconfigure Module</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+        <!-- Configuration processor -->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
```

Comment:
This exclusion for `android-json` should be added because it indeed conflicts with `com.google.cloud.ServiceOptions.getServiceAccountProjectId(ServiceOptions.java:442)` use of the `org.json.JSONTokener.<init>(Ljava/io/InputStream;)V` constructor, which doesn't exist in the stripped down `android-json` library.

Reply:
Then should we exclude `android-json` from the other sources and use the one from `spring-boot-configuration-processor`?
It's hard to fix this issue because I'm not seeing it, so I'm not sure to what extent this is happening because of your (or mine) local configuration.

In the meantime, my toy project is again picking up project ID and creds and I didn't do any change, not sure what happened but things work now...

Reply:
Maven loads dependencies in the order that they are declared. Since `spring-boot-configuration-processor` comes before `spring-cloud-gcp-core` in this POM, `JSONTokenizer` from `android-json` should get loaded. I'm not sure why you're not seeing it.

For me, actually just switching the order of `spring-boot-configuration-processor` and `spring-cloud-gcp-core` fixes the build.

Can you try running `mvn dependency:tree` on this POM to see in what order dependencies are loaded.

Reply:
I think at this point we need to configure the annotation processor [in the compiler plugin](http://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#annotationProcessorPaths). You may lose AP support in the IDE but it's better than the current arrangement IMO.

Reply:
Only `spring-boot-configuration-processor` is importing `android-json` for me.

[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ spring-cloud-gcp-autoconfigure ---
[INFO] org.springframework.cloud:spring-cloud-gcp-autoconfigure:jar:1.0.0.BUILD-SNAPSHOT
[INFO] +- org.springframework.boot:spring-boot-autoconfigure:jar:1.5.7.RELEASE:compile
[INFO] |  \- org.springframework.boot:spring-boot:jar:1.5.7.RELEASE:compile
[INFO] +- org.springframework.boot:spring-boot-configuration-processor:jar:1.5.7.RELEASE:compile
[INFO] |  \- com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:compile
[INFO] +- org.springframework.cloud:spring-cloud-gcp-core:jar:1.0.0.BUILD-SNAPSHOT:compile
[INFO] |  +- com.google.cloud:google-cloud-core:jar:1.14.0:compile
[INFO] |  |  +- com.google.guava:guava:jar:20.0:compile
[INFO] |  |  +- joda-time:joda-time:jar:2.9.2:compile
[INFO] |  |  +- org.json:json:jar:20160810:compile
[INFO] |  |  +- com.google.http-client:google-http-client:jar:1.23.0:compile
[INFO] |  |  |  \- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |  |  |     +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |  |  |     \- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.0:compile
[INFO] |  |  +- com.google.api:api-common:jar:1.2.0:compile
[INFO] |  |  +- com.google.api:gax:jar:1.15.0:compile
[INFO] |  |  |  +- com.google.auto.value:auto-value:jar:1.2:compile
[INFO] |  |  |  +- org.threeten:threetenbp:jar:1.3.3:compile
[INFO] |  |  |  \- com.google.auth:google-auth-library-oauth2-http:jar:0.7.1:compile
[INFO] |  |  |     +- com.google.auth:google-auth-library-credentials:jar:0.9.0:compile
[INFO] |  |  |     \- com.google.http-client:google-http-client-jackson2:jar:1.23.0:compile
[INFO] |  |  |        \- com.fasterxml.jackson.core:jackson-core:jar:2.8.10:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java-util:jar:3.4.0:compile
[INFO] |  |  |  +- com.google.protobuf:protobuf-java:jar:3.4.0:compile
[INFO] |  |  |  \- com.google.code.gson:gson:jar:2.8.1:compile
[INFO] |  |  +- com.google.api.grpc:proto-google-common-protos:jar:1.0.2:compile
[INFO] |  |  \- com.google.api.grpc:proto-google-iam-v1:jar:0.1.26:compile
[INFO] |  \- org.springframework:spring-core:jar:4.3.11.RELEASE:compile
[INFO] +- org.springframework:spring-web:jar:4.3.11.RELEASE:compile
[INFO] |  +- org.springframework:spring-aop:jar:4.3.11.RELEASE:compile
[INFO] |  +- org.springframework:spring-beans:jar:4.3.11.RELEASE:compile
[INFO] |  \- org.springframework:spring-context:jar:4.3.11.RELEASE:compile
[INFO] |     \- org.springframework:spring-expression:jar:4.3.11.RELEASE:compile
[INFO] +- commons-logging:commons-logging:jar:1.2:compile
[INFO] +- org.springframework.cloud:spring-cloud-context:jar:1.2.3.RELEASE:compile
[INFO] |  \- org.springframework.security:spring-security-crypto:jar:4.2.3.RELEASE:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO] +- org.mockito:mockito-core:jar:1.10.19:test
[INFO] |  \- org.objenesis:objenesis:jar:2.1:test
[INFO] \- org.springframework.boot:spring-boot-starter-test:jar:1.5.7.RELEASE:test
[INFO]    +- org.springframework.boot:spring-boot-test:jar:1.5.7.RELEASE:test
[INFO]    +- org.springframework.boot:spring-boot-test-autoconfigure:jar:1.5.7.RELEASE:test
[INFO]    +- com.jayway.jsonpath:json-path:jar:2.2.0:test
[INFO]    |  \- net.minidev:json-smart:jar:2.2.1:test
[INFO]    |     \- net.minidev:accessors-smart:jar:1.1:test
[INFO]    |        \- org.ow2.asm:asm:jar:5.0.3:test
[INFO]    +- org.assertj:assertj-core:jar:2.6.0:test
[INFO]    +- org.hamcrest:hamcrest-library:jar:1.3:test
[INFO]    +- org.skyscreamer:jsonassert:jar:1.4.0:test
[INFO]    \- org.springframework:spring-test:jar:4.3.11.RELEASE:test
[INFO] ------------------------------------------------------------------------

Reply:
@joaoandremartins Can you just switch the order of `spring-boot-configuration-processor` and `spring-cloud-gcp-core` to fix the build for me?

Reply:
I don't think you can exclude the android dependency and expect the APT to work. I believe it's no longer a dependency in Spring Boot 1.5.10 though, so upgrading to snapshots will fix it.